spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/serializer/SerializerSpecHelper.scala

```diff
@@ -0,0 +1,104 @@
+import com.intel.analytics.bigdl.utils.tf.loaders.{Pack => _}
+import com.intel.analytics.bigdl.utils.{Shape => KShape}
+import org.reflections.Reflections
+import org.reflections.scanners.SubTypesScanner
+import org.reflections.util.{ClasspathHelper, ConfigurationBuilder, FilterBuilder}
+import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers}
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+
+
+abstract class SerializerSpecHelper extends FlatSpec with Matchers with BeforeAndAfterAll{
+
+  val postFix = "bigdl"
+  val excludedClass = new mutable.HashSet[String]()
+  val excludedPackage = new mutable.HashSet[String]()
+
+  private val expected = new mutable.HashSet[String]()
+  val tested = new mutable.HashSet[String]()
+
+  protected def getPackage(): String
+
+  protected def addExcludedClass(): Unit = {}
+
+  protected def addExcludedPackage(): Unit = {}
+
+  protected def getExpected(): mutable.Set[String] = expected
+
+  override protected def beforeAll() = {
+    addExcludedClass
+    addExcludedPackage
+    val filterBuilder = new FilterBuilder()
+    excludedPackage.foreach(filterBuilder.excludePackage(_))
+    val reflections = new Reflections(new ConfigurationBuilder()
+      .filterInputsBy(filterBuilder)
+      .setUrls(ClasspathHelper.forPackage(getPackage()))
+      .setScanners(new SubTypesScanner()))
+    val subTypes = reflections.getSubTypesOf(classOf[AbstractModule[_, _, _]])
+      .asScala.filter(sub => !Modifier.isAbstract(sub.getModifiers)).
+      filter(sub => !excludedClass.contains(sub.getName))
+    subTypes.foreach(sub => expected.add(sub.getName))
+  }
+
+  protected def runSerializationTest(module : AbstractModule[_, _, Float],
+                                   input : Activity, cls: Class[_] = null) : Unit = {
+    runSerializationTestWithMultiClass(module, input,
+      if (cls == null) Array(module.getClass) else Array(cls))
+  }
+
+  protected def runSerializationTestWithMultiClass(module : AbstractModule[_, _, Float],
+                                                 input : Activity, classes: Array[Class[_]]) : Unit = {
+    val name = module.getName
+    val serFile = File.createTempFile(name, postFix)
+    val originForward = module.evaluate().forward(input)
+
+    ModulePersister.saveToFile[Float](serFile.getAbsolutePath, null, module.evaluate(), true)
+    RNG.setSeed(1000)
+    val loadedModule = ModuleLoader.loadFromFile[Float](serFile.getAbsolutePath)
+
+    val afterLoadForward = loadedModule.forward(input)
+
+    if (serFile.exists) {
+      serFile.delete
+    }
+
+    afterLoadForward should be (originForward)
+    classes.foreach(cls => tested.add(cls.getName))
+  }
+
+
+  override protected def afterAll() = {
+    var total = 0
+    getExpected().foreach(exp => {
+      require(tested.contains(exp), s" $exp not included in the test!")
+      total += 1
+    })
+    println(s"total $total, remaining ${getExpected().size - total}")
```

Comment:
should put this line before the foreach cluase otherwise it will never get executed if test for some layers are not added

Reply:
ok